### PR TITLE
OT modules calculation correction

### PR DESCRIPTION
**Pull request by piloujp**
Mainly rewrite of ot_coupon.php to be consistent with other modules and different options settings.
Bug corrections on ot_gv and ot_group_pricing.
See forum thread '[Discount coupon value](https://www.zen-cart.com/showthread.php?230003-Discount-coupon-value/page3)'.
Related to #6339

### CODE REVIEW

**Comment by drbyte** [comment]
Sorry to be nitpicky, but your text editor is using tabs instead of spaces for indentation, and thus the alignment of the code is way off all over this PR.

One example:
Screen Shot 2024-04-14 at 4 16 07 PM

**Comment by piloujp** [comment]
Sorry for that, I will try to change that. It is the automatic indentation, I think. Until I set that, I will use a regex to fix that and submit new commits.

File: includes/modules/order_total/ot_coupon.php [hunk 1]
```diff
@@ -541,39 +543,86 @@ function calculate_deductions()
                         return $od_amount;
                         break;
                     case 'P': // percentage
-//                        $od_amount['total'] = zen_round($orderTotalDetails['orderTotal']*($coupon_details['coupon_amount']/100), $currencyDecimalPlaces);
-                        $od_amount['total'] = zen_round($orderAmountTotal * ($coupon_details['coupon_amount'] / 100), $currencyDecimalPlaces);
-//                        $ratio = $od_amount['total']/$orderTotalDetails['orderTotal'];
-                        $ratio = $od_amount['total'] / $orderAmountTotal;
+                        if ($this->calculate_tax == 'Credit Note') { // Calculate deduction from gross price
+                            $od_amount['total'] = (DISPLAY_PRICE_WITH_TAX == 'true' ? $orderAmountTotal - $orderTotalDetails['orderTax'] : $orderAmountTotal) * $coupon_details['coupon_amount'] / 100;
```
Comment (lat9) [inline includes/modules/order_total/ot_coupon.php:547]: Here, and in other places, check for `DISPLAY_PRICE_WITH_TAX === 'true'`.

**Review by lat9 — CHANGES_REQUESTED** [review]
See associated comment

**Comment by piloujp** [comment]
After recently working on multiple tax rates calculation, I realized that files in this PR need more modification and these modifications can't be treated independently of my more recent code updates. I am not sure what is best, make a new PR or updating this one. I try to separate problems (and PRs) as much as possible, but It is not going to work in this case.

**Comment by piloujp** [comment]
Closing this PR replaced by PR #6432 .

**Pull request closed without merging**